messybrainz/fetch_artist_mbids.py

```
@@ -0,0 +1,99 @@
+        "artist_mbid": artist_mbid,
+        })
+
+
+def is_recording_mbid_present(connection, recording_mbid):
+    """
+        Check if recording MBID is already present in recording_artist table.
+        Returns True if recording MBID is present else False is returned.
+    """
+
+    query = text("""SELECT recording_mbid
+                      FROM recording_artist_join
+                     WHERE recording_mbid = :recording_mbid
+    """)
+
+    result = connection.execute(query, {
+    "recording_mbid": recording_mbid,
+    })
+
+    if result.rowcount:
+        return True
+
+    return False
+
+
+def fetch_and_store_artist_mbids(connection, recording_mbid):
+    """ Fetches artist MBIDs from the MusicBrainz database for the recording MBID.
+        And inserts the artist MBIDs into the recording_artist table.
+    """
+
+    recording = mb_recording.get_recording_by_mbid(recording_mbid, includes=['artists'])
+
+    artist_mbids = []
+    for artist in recording['artists']:
+        artist_mbids.append(artist['id'])
+
+    insert_artist_mbids(connection, recording_mbid, artist_mbids)
+
+
+def fetch_and_store_artist_mbids_for_all_recording_mbids(reset=False):
+    """ Fetches artist MBIDs from the musicbrainz database for the recording MBIDs
+        in the recording_json table submitted while submitting a listen.
+        Returns the number of recording MBIDs that were processed and number of
+        recording MBIDs that were added to the recording_artist table.
+    """
+
+    with db.engine.begin() as connection:
+        if reset:
+            query = text("""TRUNCATE TABLE recording_artist_join""")
+            connection.execute(query)
+
+        recording_mbids = data.fetch_distinct_recording_mbids(connection)
+        num_recording_mbids_added = 0
+        num_recording_mbids_processed = recording_mbids.rowcount
+        for recording_mbid in recording_mbids:
+            if is_valid_uuid(recording_mbid[0]):
```

Not sure if this step is even needed -- we can't insert invalid UUIDs into the database... But, if we keep it, then we should at least report an error to the user.

---

These UUIDs are inserted in JSON fields of recording_json table. And we don't have any check in messybrainz submit_listen to check if recording_mbid is valid UUID or not. But we get recordings only from LB which checks for valid UUIDs. So, as you said it's not possible to have invalid UUIDs. I'll remove this check. And in case we need it we should add it to submit_listen function.